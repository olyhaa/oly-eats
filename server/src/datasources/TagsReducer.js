export const tagReducer = (tag) => {
  if (!tag) {
    return null;
  }
  return {
    id: tag.id,
    typeid: tag.typeid,
    label: tag.label,
  };
};

export const tagMutationReducer = ({
  success = false,
  message = undefined,
  tag = null,
}) => {
  return {
    success,
    message,
    tag: tagReducer(tag),
  };
};
