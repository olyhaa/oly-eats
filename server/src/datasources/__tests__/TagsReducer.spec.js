import { tagReducer, tagMutationReducer } from '../TagsReducer';

const mockTags = [
  {
    id: '1',
    typeid: '1',
    label: 'bakery',
  },
  {
    id: '2',
    typeid: '1',
    label: 'casserole',
  },
  {
    id: '3',
    typeid: '1',
    label: 'salad',
  },
  {
    id: '7',
    typeid: '2',
    label: 'american',
  },
  {
    id: '8',
    typeid: '2',
    label: 'asian',
  },
];

describe('tag reducer', () => {
  it('properly transforms tag object', () => {
    expect(tagReducer(mockTags[0])).toMatchSnapshot();
  });
});

describe('tag mutation reducer', () => {
  it('properly transforms tag object', () => {
    expect(tagMutationReducer({ tag: mockTags[0] })).toMatchSnapshot();
  });

  it('properly transforms tag type object on success', () => {
    expect(
      tagMutationReducer({
        success: true,
        tag: mockTags[0],
      })
    ).toMatchSnapshot();
  });

  it('properly transforms tag type object on success with message', () => {
    expect(
      tagMutationReducer({
        success: true,
        message: 'great job',
        tag: mockTags[0],
      })
    ).toMatchSnapshot();
  });

  it('properly transforms on success with message', () => {
    expect(
      tagMutationReducer({
        success: true,
        message: 'great job',
      })
    ).toMatchSnapshot();
  });

  it('properly transforms on failure', () => {
    expect(tagMutationReducer({ success: false })).toMatchSnapshot();
  });

  it('properly transforms on failure with message', () => {
    expect(
      tagMutationReducer({ success: false, message: 'bad test' })
    ).toMatchSnapshot();
  });
});
