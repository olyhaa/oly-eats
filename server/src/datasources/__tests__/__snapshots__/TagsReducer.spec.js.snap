// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tag mutation reducer properly transforms on failure 1`] = `
Object {
  "message": undefined,
  "success": false,
  "tag": null,
}
`;

exports[`tag mutation reducer properly transforms on failure with message 1`] = `
Object {
  "message": "bad test",
  "success": false,
  "tag": null,
}
`;

exports[`tag mutation reducer properly transforms on success with message 1`] = `
Object {
  "message": "great job",
  "success": true,
  "tag": null,
}
`;

exports[`tag mutation reducer properly transforms tag object 1`] = `
Object {
  "message": undefined,
  "success": false,
  "tag": Object {
    "id": "1",
    "label": "bakery",
    "typeid": "1",
  },
}
`;

exports[`tag mutation reducer properly transforms tag type object on success 1`] = `
Object {
  "message": undefined,
  "success": true,
  "tag": Object {
    "id": "1",
    "label": "bakery",
    "typeid": "1",
  },
}
`;

exports[`tag mutation reducer properly transforms tag type object on success with message 1`] = `
Object {
  "message": "great job",
  "success": true,
  "tag": Object {
    "id": "1",
    "label": "bakery",
    "typeid": "1",
  },
}
`;

exports[`tag reducer properly transforms tag object 1`] = `
Object {
  "id": "1",
  "label": "bakery",
  "typeid": "1",
}
`;
